File: 20190519python/0189. Rotate Array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        k=k%len(nums)
        if k!=0:
            lenOfNums=len(nums)
            nums[:lenOfNums-k]=nums[:lenOfNums-k][::-1]
            nums[-k:]=nums[-k:][::-1]
            nums[:]=nums[:][::-1]

File: archives/C#/0189. Rotate Array.cs
public class Solution {
    public void Rotate(int[] nums, int k) {
        k=k%nums.Length;
        Reverse(nums,0,nums.Length-k-1);
        Reverse(nums,nums.Length-k,nums.Length-1);
        Reverse(nums,0,nums.Length-1);
        //return nums;
        
    }
    public int[] Reverse(int[] nums,int left,int right){
        while(left<right){
            int tmp=nums[left];
            nums[left]=nums[right];
            nums[right]=tmp;
            left+=1;
            right-=1;
        }
        return nums;
    }
}
